### ISSUE TITLE
no-shadowed-variable false positive on inferred conditional type

### ISSUE THREAD

**Issue by pedro-pedrosa** [palantir/tslint#4235]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.3
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
interface I1<T, U> {
    t: T;
    u: U;
}
type T0<T> =
    T extends I1<infer Tinfer, number> ?
        number :
        T extends I1<infer Tinfer, string> ?
            string :
            never;
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint:recommended"],
    "rules": {}
}
```

#### Actual behavior
`ERROR: 8:28  no-shadowed-variable  Shadowed name: 'Tinfer'`

#### Expected behavior
Inferred types on conditional type expressions only exist in the true branch of the conditional. Tslint shouldn't error here.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
